RELEASE_NOTES.md

```diff
@@ -0,0 +1,69 @@
+- feat: Add a `/verificationLogger:csv` CLI option that emits the same status and timing information as `/verificationLogger:trx`, but in an easier-to-parse format, along with Z3 resource counts for more repeatable tracking of verification difficulty.
+
+### IDE 
+- fix: Clean up process resources if IDE closed or restarted.
+- fix: Do not let the Dafny compilation status bar get in a stuck state.
+
+- feat: Verification status reporting shows which proof is being verified, which can help debug slow to verify proofs.
+- feat: Publish parsing and resolution diagnostics before verification has completed. Verification diagnostics from previous runs are migrated.
+- feat: Enable 'go to definition', 'hover' and 'signature help' features before verification has completed.
+- feat: Improve the hover feature to work for a wider scope of Dafny constructs, including function and method parameters, forall, exists and let expressions, and set and map comprehensions.
+- feat: Add an experimental verification caching feature, which enables automatically determining which proofs need to verify again after making changes.
+- feat: Display related resolution errors using nested diagnostics instead of independent diagnostics.
+
+### UX
+- feat: Improve error reporting when providing incorrectly typed arguments in a function call.
+- feat: Improve error reporting when using type tests.
+
+### C#
+- fix: resolve various instances where Dafny would produce invalid C# code (#1607, #1761, and #1762).
+- feat: Support variant type parameters on datatype definitions, which enables using traits as type arguments (https://github.com/dafny-lang/dafny/issues/1499).
+- feat: Support for downcasting both custom datatypes and functions.
+
+### Various improvements
+- fix: `DafnyLanguageServer.dll` and `Dafny.dll` depended on two different versions of Newtonsoft.Json, which could cause crashes in development environments.
+- fix: (error reporting) Types with the same name but from different modules are now disambiguated in error messages.
+- fix: (error reporting) Messages about arguments / parameters type mismatch are clearer and include the parameter name if available.
+- fix: (robustness) Exceptions during parsing, if any, won't crash the language server anymore.
+- fix: The elephant operator (`:-`) has a clearer error message and no longer reject generic methods on its right-hand side.
+
+## Breaking changes
+
+- Proofs such as methods and lemmas whose verification behavior, either the result or the verification time, depend on arbitrary behavior in Dafny's solver, may show different verification behavior in Dafny 3.4. For users who are not ready to change those proofs to make their verification more reliable, we offer the command line option `/mimicVerificationOf:3.3` to keep the Dafny 3.4 verification behavior consistent with 3.3.
```

Comment:
```suggestion
- The verifier in Dafny 3.4 is now more efficient for many programs, and making changes to Dafny programs is less likely to cause verification to take longer or timeout. However, it is still possible for some correct programs to take longer to verify than on Dafny 3.3, or for verification to fail. For users with such programs who are not yet ready to modify them to pass the 3.4 verifier, we offer the command line option `/mimicVerificationOf:3.3` to keep the Dafny 3.4 verification behavior consistent with 3.3.
```

Saying the verification result can depend on arbitrary behavior in Dafny's solver seems to imply that invalid code might successfully verify.